examples/platform/silabs/matter_aws/matter_aws_interface/src/MatterAws.cpp

```diff
@@ -83,8 +83,12 @@ static void MatterAwsMqttConnCb(mqtt_client_t * client, void * arg, mqtt_connect
     ChipLogProgress(AppServer, "[MATTER_AWS] MQTT connection status: %u", status);
     if (status != MQTT_CONNECT_ACCEPTED)
     {
-        if (matterAwsTask)
+        /* Signal the task to exit cleanly instead of setting flag only */
+        if (matterAwsTask != NULL && matterAwsEvents != NULL)

```

Comment:
- as i understand from the sequence of initialization, if matterAwsTask is not NULL, then matterAwsEvents cannot be NULL. So, any specific reason for adding this additional condition?